test/Dialect/FIRRTL/canonicalization.mlir

```diff
@@ -3576,19 +3576,38 @@ firrtl.module private @Issue7562(in %sel : !firrtl.uint<1>, in %a : !firrtl.cons
 }
 
 // CHECK-LABEL: firrtl.class @PropertyArithmetic
-firrtl.class @PropertyArithmetic(in %in: !firrtl.integer, out %out0: !firrtl.integer, out %out1: !firrtl.integer) {
+firrtl.class @PropertyArithmetic(in %in: !firrtl.integer, out %out0: !firrtl.integer, out %out1: !firrtl.integer, out %out2: !firrtl.integer, out %out3: !firrtl.integer, out %out4: !firrtl.integer, out %out5: !firrtl.integer) {
   // CHECK: [[C4:%.+]] = firrtl.integer 4
+  // CHECK: [[C10:%.+]] = firrtl.integer 10
+  // CHECK: [[C3:%.+]] = firrtl.integer 3
   %0 = firrtl.integer 0
   %1 = firrtl.integer 1
   %2 = firrtl.integer 2
+  %5 = firrtl.integer 5
 
-  %3 = firrtl.integer.shl %1, %2 : (!firrtl.integer, !firrtl.integer) -> !firrtl.integer
-  %4 = firrtl.integer.shl %in, %0 : (!firrtl.integer, !firrtl.integer) -> !firrtl.integer
+  %res0 = firrtl.integer.add %1, %2 : (!firrtl.integer, !firrtl.integer) -> !firrtl.integer
+  %res1 = firrtl.integer.add %in, %0 : (!firrtl.integer, !firrtl.integer) -> !firrtl.integer
 
-  // CHECK: firrtl.propassign %out0, [[C4]]
+  // CHECK: firrtl.propassign %out0, [[C3]]
   // CHECK: firrtl.propassign %out1, %in
-  firrtl.propassign %out0, %3 : !firrtl.integer
-  firrtl.propassign %out1, %4 : !firrtl.integer
+  firrtl.propassign %out0, %res0 : !firrtl.integer
+  firrtl.propassign %out1, %res1 : !firrtl.integer
+
+  %res2 = firrtl.integer.mul %2, %5 : (!firrtl.integer, !firrtl.integer) -> !firrtl.integer
+  %res3 = firrtl.integer.mul %in, %1 : (!firrtl.integer, !firrtl.integer) -> !firrtl.integer
+
+  // CHECK: firrtl.propassign %out2, [[C10]]
+  // CHECK: firrtl.propassign %out3, %in
+  firrtl.propassign %out2, %res2 : !firrtl.integer
+  firrtl.propassign %out3, %res3 : !firrtl.integer
+
+  %res4 = firrtl.integer.shl %1, %2 : (!firrtl.integer, !firrtl.integer) -> !firrtl.integer
+  %res5 = firrtl.integer.shl %in, %0 : (!firrtl.integer, !firrtl.integer) -> !firrtl.integer
```

Comment:
nit: it made the diff more confusing to move these to the bottom and rename their SSA values. At first, I thought "where'd the shl folder tests go".

I do like the name better, but in this case I think we'd prefer a smaller diff. Specifically, I think a couple LLVM guidelines apply here:

https://llvm.org/docs/CodingStandards.html#introduction (the note about sticking to existing style)
https://llvm.org/docs/DeveloperPolicy.html#incremental-development (each change should be kept as small as possible)

In this case, it should be possible to update this test file by adding new outputs and operations, without changing existing lines besides the module signature.

Reply:
Right, there's no need to enforce any particular order of these tests. Done! And I only renames result values for `shl`.